src/DBmysqlIterator.php

```diff
@@ -703,17 +711,21 @@ private function analyseFkey($values)
                 $f1 = $values[$t1];
                 $t2 = $keys[1];
                 $f2 = $values[$t2];
-                if ($f2 instanceof QuerySubQuery) {
+                if ($f2 instanceof QuerySubQuery || $f2 instanceof QueryExpression) {
                     return (is_numeric($t1) ? DBmysql::quoteName($f1) : DBmysql::quoteName($t1) . '.' . DBmysql::quoteName($f1)) . ' = ' .
-                    $f2->getQuery();
+                    $f2;
```

Comment:
Both QuerySubQuery and QueryExpression implement the toString method now so there is no need to call the class-specific methods.